import_export/admin.py

```diff
@@ -731,6 +732,21 @@ def export_action(self, request):
         if request.POST and "export_items" in request.POST:
             # this field is instantiated if the export is POSTed from the
             # 'action' drop down
+
+            if (
+                type(self).get_valid_export_item_pks
+                != ExportMixin.get_valid_export_item_pks
+            ):
+                warnings.warn(
```

Comment:
Should we move this warning to inside `get_valid_export_item_pks()` ?

I think it is ok just to say it is 'deprecated and will be removed in a future release'.  The release notes can explain why and what users should do.

Reply:
That's actually the first thing I tried, but I didn't like it, because it issues the warning when the method isn't overwritten. This is bad because it breaks tests and issues the warning to users who are not using the method and doesn't issue it in the case the method is overwritten. Should I add a comment clarifying this?

I can remove the verbose message.

Reply:
I note that django will place their deprecation warnings in the method ([example](https://github.com/django/django/blob/0fdcf1029cea2d43bd68c5270f48e0f7deab5e47/django/db/models/query.py#L2241)), so I have no problem with putting the deprecation warning in the method.  It would be my preference because it's what we do elsewhere in the code base.

A small number of users will see the warning if they run with deprecation warnings enabled, which is a good thing, and why we have the warning.  They will see it in their own implementation if they call super(), otherwise they will see it in the release notes.  So I think this is ok as well.

Re the tests, you can add a decorator to ignore the warning (see the pattern [here](https://github.com/django-import-export/django-import-export/blob/06557c3e392b77e36c34db0940be0cfbaddc55c2/tests/core/tests/utils.py#L5)).

I would probably prefer to have the warning inside the method, but I think it is ok how you have done it.

Reply:
Hi, I modified the PR along the lines I described above.  I understand your reasoning, but my preference is to be consistent with the other deprecations in the project.

Reply:
Hi Matt, sorry that I have not replied. I completely agree with your reasoning, consistency with the rest of the codebase and django is paramount, and I am only learning, so it is better to follow professional standards. I am sorry I could not find the time to do the requested modifications. Thank you for your efforts!

Reply:
No problem at all, thanks for all your help with this PR, much appreciated.  I won't merge just yet as we have a few more changes planned before 4.2